### ISSUE TITLE
Integration stopped working - EU - Invalid Application/Device Id

### ISSUE THREAD

**Issue by tobixx** [Hyundai-Kia-Connect/kia_uvo#660]
**Region and Brand of car**
EU/KIA/EV6

**Version of the integration**
2.10.2

**Describe the bug**
Can't connect anymore. Started just this night after months of no issues. No other changes were done.

**Debug logs if an error occurred**

```
Logger: homeassistant.components.websocket_api.http.connection
Source: custom_components/kia_uvo/coordinator.py:151
Integration: Home Assistant WebSocket API (documentation, issues)
First occurred: 06:29:50 (3 occurrences)
Last logged: 09:18:43

[546552013968] Server returned: 'Invalid request value - Invalid Application Id.'
[546910811920] Server returned: 'Invalid request body - Invalid deviceId. '
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/components/websocket_api/commands.py", line 205, in handle_call_service
    await hass.services.async_call(
  File "/usr/src/homeassistant/homeassistant/core.py", line 1910, in async_call
    task.result()
  File "/usr/src/homeassistant/homeassistant/core.py", line 1950, in _execute_service
    await cast(Callable[[ServiceCall], Awaitable[None]], handler.job.target)(
  File "/config/custom_components/kia_uvo/services.py", line 54, in async_handle_update
    await coordinator.async_update_all()
  File "/config/custom_components/kia_uvo/coordinator.py", line 151, in async_update_all
    await self.hass.async_add_executor_job(
  File "/usr/local/lib/python3.11/concurrent/futures/thread.py", line 58, in run
    result = self.fn(*self.args, **self.kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/hyundai_kia_connect_api/VehicleManager.py", line 75, in update_all_vehicles_with_cached_state
    self.update_vehicle_with_cached_state(vehicle_id)
  File "/usr/local/lib/python3.11/site-packages/hyundai_kia_connect_api/VehicleManager.py", line 80, in update_vehicle_with_cached_state
    self.api.update_vehicle_with_cached_state(self.token, vehicle)
  File "/usr/local/lib/python3.11/site-packages/hyundai_kia_connect_api/KiaUvoApiEU.py", line 296, in update_vehicle_with_cached_state
    state = self._get_cached_vehicle_state(token, vehicle)
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/hyundai_kia_connect_api/KiaUvoApiEU.py", line 710, in _get_cached_vehicle_state
    _check_response_for_errors(response)
  File "/usr/local/lib/python3.11/site-packages/hyundai_kia_connect_api/KiaUvoApiEU.py", line 110, in _check_response_for_errors
    raise APIError(f"Server returned: '{response['resMsg']}'")
hyundai_kia_connect_api.exceptions.APIError: Server returned: 'Invalid request value - Invalid Application Id.'
```

**To Reproduce**
Nothing special. Working integration just stopped working this night.

**Expected behavior**
Well, that it can connect.

**Comment by compuvision2** [comment]
This behaviour is also with my Ioniq 5 since last night.

**Comment by vahaldor** [comment]
Same issue here. KIA SORENTO PHEV

**Comment by MYG63** [comment]
I think I have same problem here.

2023-06-29 09:36:44.964 DEBUG (SyncWorker_10) [hyundai_kia_connect_api.KiaUvoApiEU] hyundai_kia_connect_api - get_stamp 2023-06-29 05:02:58.607000+00:00 120000 76 998 76.88631513333333
2023-06-29 09:37:00.123 ERROR (MainThread) [custom_components.kia_uvo.config_flow] Unexpected exception
Traceback (most recent call last):
  File "/config/custom_components/kia_uvo/config_flow.py", line 159, in async_step_user
    await validate_input(self.hass, user_input)
  File "/config/custom_components/kia_uvo/config_flow.py", line 63, in validate_input
    token: Token = await hass.async_add_executor_job(
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/concurrent/futures/thread.py", line 58, in run
    result = self.fn(*self.args, **self.kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/hyundai_kia_connect_api/KiaUvoApiEU.py", line 198, in login
    device_id = self._get_device_id(stamp)
                ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/hyundai_kia_connect_api/KiaUvoApiEU.py", line 1104, in _get_device_id
    _check_response_for_errors(response)
  File "/usr/local/lib/python3.11/site-packages/hyundai_kia_connect_api/KiaUvoApiEU.py", line 108, in _check_response_for_errors
    raise error_code_mapping[response["resCode"]](response["resMsg"])
hyundai_kia_connect_api.exceptions.RequestTimeoutError: Undefined Error - notNull Violation: devices.device_id cannot be null

**Comment by DickiePhitt** [comment]
Same problem. All entities showing unavailable.

**Comment by Lesterpig** [comment]
I'm having the same issue. Relevant issue in the underlying API wrapper: https://github.com/Hyundai-Kia-Connect/hyundai_kia_connect_api/issues/348

**Comment by MYG63** [comment]
I think the reason for this could be that the Kia cloud server is not available. When adding the integration and checking the logfile in parallel with "tail -f" it needs some time - about 5 to 10 seconds" after the "get_stamp" line. That could be an indicator for a non existing answer from some server or a server which does not respond at all.

**Comment by compuvision2** [comment]
I tested the connection to to the hyundai server with the bluelink app.
There was a normal response from this side.

MYG63 ***@***.***> schrieb am Do., 29. Juni 2023, 10:21:

**Comment by vasil-github** [comment]
Same problem here.
After enjoying this integration for 9 days, today it stopped working and when I tried to readd the integration the error nvalid Application Id...... Popped up.
Hyundai EU user

**Comment by Rothammel** [comment]
Same Here, also the Power App from tibber is not working 😂

**Comment by DragonQ** [comment]
Same here.

**Comment by bumperboy** [comment]
Same problem here. Home assistant is not receiving the information anymore.

Thank you for your help in fixing this.

**Comment by peter4200** [comment]
Same problem here. 
Ioniq 5 placed in Denmark

**Comment by fuatakgun** [comment]
Please let's keep the comment stream clean as much as possible so in case something important pops up, we don't miss it.

I am working on this and trying to figure out what has changed in the login flow.

Also checking other repositories if they have a solution in place.

`Me too` style comments are not helpful, while i understand the frustration here, let's be patient.

**Comment by DutchessNicole** [comment]
I have a similar issue since 2 days, however I get a different error message:

```
2023-07-01 08:44:50.731 WARNING (MainThread) [hyundai_kia_connect_api.KiaUvoApiEU] Unsupported language: en-GB, fallback to en
2023-07-01 08:44:51.058 DEBUG (SyncWorker_34) [hyundai_kia_connect_api.KiaUvoApiEU] hyundai_kia_connect_api - get_stamp 2023-07-01 05:01:41.828000+00:00 120000 51 998 51.57691609166667
2023-07-01 08:44:51.298 ERROR (MainThread) [custom_components.kia_uvo.coordinator] Unexpected error fetching kia_uvo data: Server returned: 'Invalid request body - Invalid parameter.'
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/update_coordinator.py", line 283, in _async_refresh
    self.data = await self._async_update_data()
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/kia_uvo/coordinator.py", line 107, in _async_update_data
    await self.async_check_and_refresh_token()
  File "/config/custom_components/kia_uvo/coordinator.py", line 166, in async_check_and_refresh_token
    await self.hass.async_add_executor_job(
  File "/usr/local/lib/python3.11/concurrent/futures/thread.py", line 58, in run
    result = self.fn(*self.args, **self.kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/hyundai_kia_connect_api/VehicleManager.py", line 120, in check_and_refresh_token
    self.initialize()
  File "/usr/local/lib/python3.11/site-packages/hyundai_kia_connect_api/VehicleManager.py", line 64, in initialize
    self.token: Token = self.api.login(self.username, self.password)
                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/hyundai_kia_connect_api/KiaUvoApiEU.py", line 198, in login
    device_id = self._get_device_id(stamp)
                ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/hyundai_kia_connect_api/KiaUvoApiEU.py", line 1104, in _get_device_id
    _check_response_for_errors(response)
  File "/usr/local/lib/python3.11/site-packages/hyundai_kia_connect_api/KiaUvoApiEU.py", line 110, in _check_response_for_errors
    raise APIError(f"Server returned: '{response['resMsg']}'")
hyundai_kia_connect_api.exceptions.APIError: Server returned: 'Invalid request body - Invalid parameter.'
2023-07-01 08:44:51.315 DEBUG (MainThread) [custom_components.kia_uvo.coordinator] Finished fetching kia_uvo data in 0.577 seconds (success: False)
```
Debug logging doesn't really seem to add any interesting information, even from the API debug logs. It might be interesting to see what the request body now actually is to figure out what the API is choking on?

**Comment by stephenmhall** [comment]
For info my Android UVO app isn't working either, coming up with a network error popup

**Comment by itsaints** [comment]
> 

Some here, Integration stopped working then on checking UVO app now not working and kia website pointing to Kia Connect as the replacement. Kia Connect installed and working but assume API endpoints may have changed given app changed.

**Comment by SectorTll** [comment]
Unexpected error fetching kia_uvo data: Server returned: 'Invalid request body - Invalid parameter.'
Traceback (most recent call last):

**Comment by caddyblack** [comment]
Same server issue here after HA update. Tried a re-config of the integration when I noticed the error. From that re-config:
Is this problem linked to the extra configuration parameter 'pin' ? I can't remember having that in the config before.

**Comment by smalarz** [comment]
Same here. EU/KIA/E-NIRO

**Comment by Anobium** [comment]
Same here.  UK/HY/IONIQ5_23

**Comment by probize** [comment]
Same here IONIQ5_23, tried to remove and reinstall but authentication fails... It was better just to have left it I assume

**Comment by pyrocumulus** [comment]
Same problem here unfortunately, EU, Hyundai Ioniq 5 (2023).

**Comment by seiferrtalle** [comment]
Unfortunately same problem here. EU. Kia XCeed (2019)

Logger: custom_components.kia_uvo.coordinator
Source: custom_components/kia_uvo/coordinator.py:166
Integration: Hyundai / Kia Connect (documentation, issues)
First occurred: 1. Juli 2023 um 06:52:21 (1122 occurrences)
Last logged: 07:54:25

Unexpected error fetching kia_uvo data: Server returned: 'Invalid request body - Invalid parameter.'
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/update_coordinator.py", line 283, in _async_refresh
    self.data = await self._async_update_data()
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/kia_uvo/coordinator.py", line 107, in _async_update_data
    await self.async_check_and_refresh_token()
  File "/config/custom_components/kia_uvo/coordinator.py", line 166, in async_check_and_refresh_token
    await self.hass.async_add_executor_job(
  File "/usr/local/lib/python3.11/concurrent/futures/thread.py", line 58, in run
    result = self.fn(*self.args, **self.kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/hyundai_kia_connect_api/VehicleManager.py", line 120, in check_and_refresh_token
    self.initialize()
  File "/usr/local/lib/python3.11/site-packages/hyundai_kia_connect_api/VehicleManager.py", line 64, in initialize
    self.token: Token = self.api.login(self.username, self.password)
                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/hyundai_kia_connect_api/KiaUvoApiEU.py", line 198, in login
    device_id = self._get_device_id(stamp)
                ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/hyundai_kia_connect_api/KiaUvoApiEU.py", line 1104, in _get_device_id
    _check_response_for_errors(response)
  File "/usr/local/lib/python3.11/site-packages/hyundai_kia_connect_api/KiaUvoApiEU.py", line 110, in _check_response_for_errors
    raise APIError(f"Server returned: '{response['resMsg']}'")
hyundai_kia_connect_api.exceptions.APIError: Server returned: 'Invalid request body - Invalid parameter.'

**Comment by andig** [comment]
Is it really that difficult: https://github.com/Hyundai-Kia-Connect/kia_uvo/issues/660#issuecomment-1615157981

**Comment by rikardkjell** [comment]
> For info my Android UVO app isn't working either, coming up with a network error popup.
> 
> The UVO app has gone from the play store, there is now just Kia Connect which works.

Confirmed by Kia support when I contacted them that the Kia UVO app have been closed down. The Kia Connect is the app to use.

**Comment by crackers81** [comment]
> > For info my Android UVO app isn't working either, coming up with a network error popup.
> > The UVO app has gone from the play store, there is now just Kia Connect which works.
> 
> Confirmed by Kia support when I contacted them that the Kia UVO app have been closed down. The Kia Connect is the app to use.
The Connect app has been the goto app for 1 year atleast.

**Comment by cdnninja** [comment]
We are getting close to an update on this everyone.  Thanks for the patience.   Hope to have this working in the next few days.

**Comment by HansNetcom** [comment]
Same problem with my Kia eNero. Worked perfect in the past and suddenly stops working.
Hope to have a working update soon. Thanx in advance

**Comment by cdnninja** [comment]
The "master" branch has an update that may help Hyundai users.  Another item is needed for Kia users for sure.

**Comment by ronanmu** [comment]
Latest patch confirmed as working for UK IONIQ5 MY23 🥳 
Thanks for all such a quick turnaround on this 👍🏻

**Comment by Bluhme1** [comment]
Confirmed working in Denmark Hyundai Kona.
Thanks a lot guys. 

I wonder when Hyundai /Kia will stop this "cat and mouse" game

**Comment by CBrosius** [comment]
>  Another item is needed for Kia users for sure.

Do you need a Kia user to get this Item or is this already found? 

If still needed, I can try to get the Info If you guide me.

**Comment by cdnninja** [comment]
> > Another item is needed for Kia users for sure.
> 
> Do you need a Kia user to get this Item or is this already found?
> 
> If still needed, I can try to get the Info If you guide me.

We have it but it is in another repo and need that owner to merge.

**Comment by RuneNyhuus** [comment]
Fix is working in Denmark for Ioniq 5 aswell.

**Comment by fuatakgun** [comment]
Problem should be fixed for Hyundai users, at least as of now.
For Kia, the expected files are generated but the issue still persists.

**Comment by richsimpson** [comment]
Great work to solve it for many. Working by installing the master (didn't realise I was using a Beta) from HACS on a 2022 UK Hyundai. 

In HACS, integrations then open the kia/hyundai intergration. Select re-download from the top right menu icon of HACS in HASS. Then select master. With a restart of HASS afterwards for good measure.

Cheers

**Comment by rikardkjell** [comment]
As of now when trying to add integration...
EU/KIA/E-Niro

Logger: custom_components.kia_uvo.config_flow
Source: custom_components/kia_uvo/config_flow.py:63
Integration: Kia Uvo / Hyundai Bluelink (documentation, issues)
First occurred: 12:38:22 (3 occurrences)
Last logged: 13:10:52

Unexpected exception
Traceback (most recent call last):
  File "/config/custom_components/kia_uvo/config_flow.py", line 159, in async_step_user
    await validate_input(self.hass, user_input)
  File "/config/custom_components/kia_uvo/config_flow.py", line 63, in validate_input
    token: Token = await hass.async_add_executor_job(
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/concurrent/futures/thread.py", line 58, in run
    result = self.fn(*self.args, **self.kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/hyundai_kia_connect_api/KiaUvoApiEU.py", line 198, in login
    device_id = self._get_device_id(stamp)
                ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/hyundai_kia_connect_api/KiaUvoApiEU.py", line 1104, in _get_device_id
    _check_response_for_errors(response)
  File "/usr/local/lib/python3.11/site-packages/hyundai_kia_connect_api/KiaUvoApiEU.py", line 110, in _check_response_for_errors
    raise APIError(f"Server returned: '{response['resMsg']}'")
hyundai_kia_connect_api.exceptions.APIError: Server returned: 'Invalid request body - Invalid parameter.'

**Comment by rikl64** [comment]
With the latest patch (api version 3.3.8) for a Kia EV6 UK - HA log gives:
Traceback (most recent call last):
  File "/config/custom_components/kia_uvo/config_flow.py", line 159, in async_step_user
    await validate_input(self.hass, user_input)
  File "/config/custom_components/kia_uvo/config_flow.py", line 63, in validate_input
    token: Token = await hass.async_add_executor_job(
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/concurrent/futures/thread.py", line 58, in run
    result = self.fn(*self.args, **self.kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/hyundai_kia_connect_api/KiaUvoApiEU.py", line 199, in login
    device_id = self._get_device_id(stamp)
                ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/hyundai_kia_connect_api/KiaUvoApiEU.py", line 1107, in _get_device_id
    _check_response_for_errors(response)
  File "/usr/local/lib/python3.11/site-packages/hyundai_kia_connect_api/KiaUvoApiEU.py", line 111, in _check_response_for_errors
    raise APIError(f"Server returned: '{response['resMsg']}'")
hyundai_kia_connect_api.exceptions.APIError: Server returned: 'Invalid request body - Block the UVO app id.'

**Comment by paulschermers** [comment]
That is expected for the time being @rikl64 and @rikardkjell as it is only resolved for Hyundai. Not yet for **Kia**.
Please see the message of @fuatakgun earlier

> Problem should be fixed for Hyundai users, at least as of now. For Kia, the expected files are generated but the issue still persists.

**Comment by pa5cal84** [comment]
Same here, i think:

Deze fout is ontstaan door een aangepaste integratie.

Logger: custom_components.kia_uvo.coordinator
Source: custom_components/kia_uvo/coordinator.py:166
Integration: Kia Uvo / Hyundai Bluelink (documentation, issues)
First occurred: 29 juni 2023 om 16:04:12 (4301 occurrences)
Last logged: 16:01:33

Unexpected error fetching kia_uvo data: Server returned: 'Invalid request body - Invalid parameter.'
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/update_coordinator.py", line 283, in _async_refresh
    self.data = await self._async_update_data()
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/kia_uvo/coordinator.py", line 107, in _async_update_data
    await self.async_check_and_refresh_token()
  File "/config/custom_components/kia_uvo/coordinator.py", line 166, in async_check_and_refresh_token
    await self.hass.async_add_executor_job(
  File "/usr/local/lib/python3.11/concurrent/futures/thread.py", line 58, in run
    result = self.fn(*self.args, **self.kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/hyundai_kia_connect_api/VehicleManager.py", line 120, in check_and_refresh_token
    self.initialize()
  File "/usr/local/lib/python3.11/site-packages/hyundai_kia_connect_api/VehicleManager.py", line 64, in initialize
    self.token: Token = self.api.login(self.username, self.password)
                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/hyundai_kia_connect_api/KiaUvoApiEU.py", line 198, in login
    device_id = self._get_device_id(stamp)
                ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/hyundai_kia_connect_api/KiaUvoApiEU.py", line 1104, in _get_device_id
    _check_response_for_errors(response)
  File "/usr/local/lib/python3.11/site-packages/hyundai_kia_connect_api/KiaUvoApiEU.py", line 110, in _check_response_for_errors
    raise APIError(f"Server returned: '{response['resMsg']}'")
hyundai_kia_connect_api.exceptions.APIError: Server returned: 'Invalid request body - Invalid parameter.'

**Comment by fuatakgun** [comment]
please try again with latest version, it works locally and cloud tests for Kia EU.

**Comment by raducotescu** [comment]
You can close this one - https://github.com/Hyundai-Kia-Connect/kia_uvo/pull/667 seems to have fixed it. Thanks a lot for your hard work! :)

**Comment by Bluhme1** [comment]
No, no, stopped working in Denmark for Hyundai Kona

**Comment by raducotescu** [comment]
Sorry, I tested with my Kia. Indeed, it seems that the fix for the Kia broke the fix for Hyundai. See https://github.com/Hyundai-Kia-Connect/hyundai_kia_connect_api/issues/354.

**Comment by rikardkjell** [comment]
Confirmed. Recent Master now work for Kia...

**Comment by HansNetcom** [comment]
Well done, works again like a charm. Thank you so much :-)

**Comment by Bluhme1** [comment]
Will there be a new version (2.10.4) with the new patch for both Kia and Hyundai?

**Comment by Taomyn** [comment]
Damn, the first fix worked fine for my Hyundai, then I saw the notification for 2.0.3 and stupidly thinking it was just the official patch applied it.

Now it's stopped working like before. Can't revert, even tried re-applying master, rebooted/shutdown both HA and the HA OS, nothing fixes it again.

**Comment by cdnninja** [comment]
Yes.  Master branch now has that version.   I am not by a pc so can't generate the release package.   I'll do that today though.

**Comment by cdnninja** [comment]
> 0. Don't want to spam, but have to say:
> 
> 1. Many, many thanks to all contributors! (works for me on master).
> 
> 2. Just to let you know why, for example, your work is so important (at least for me) : I use it in combination with the "EV smart charging" integration for, well, charging my car automatically according to the "nordpool" integration - which means charging when most renewable power is available in the grid (and to whom it matters: the most cost savings way)
> 
> 3. Just a thought how to incorporate more contributors (for the next issue): a guide on the sniffing setup in the readme would be helpful (for different platforms). Maybe one of the guys with a working setup could contribute that ...

Thanks for the feedback!   Glad this can be helpful.  

As for more contributors the sniffing setup unfortunately keeps changing as they try to cut us out.   Right now the conversations on this happen in the discord for blue linky.   We work together when this breaks.   Always welcome more help on this integration, the underlying api and I am sure the bluelinky team are the same! 

A how to sniff does exist in our wiki here but it doesn't work for EU just other regions.

**Comment by Taomyn** [comment]
v2.0.4 confirmed working for me again, Hyundai EU, and no more memory errors, thank-you so much

**Comment by HansNetcom** [comment]
v2.0.4 confirmed working for KIA eNero EU 👍🙏

**Comment by Rothammel** [comment]
v2.10.4 is working now, awesome my friend

**Comment by seiferrtalle** [comment]
v2.10.4 works great with my KIA XCeed 2019! 😀Also from my side many thanks to all for the hard work and super fast fix!

**Comment by bootc** [comment]
I've opened a new issue (https://github.com/Hyundai-Kia-Connect/kia_uvo/issues/672) because this isn't working again for me (Kia, EU).

**Comment by crackers81** [comment]
> I've opened a new issue (#672) because this isn't working again for me (Kia, EU).

Then the problem is local. Kia Norway here. Working like a charm :)

**Comment by DickiePhitt** [comment]
Also still got the problem.

**Comment by Taomyn** [comment]
No issues here, Hyundai, EU, Luxembourg, did a forced update and got everything as expected, nothing logged

**Comment by HansNetcom** [comment]
sorry to say, I got this problem after the update.
Tried to unlock the door, it worked, then to lock the door, I got this message:

Filed to call service lock/lock. Server returned: ‘Invalid request body - invalid device ID’

**Comment by BjoernWaechter** [comment]
> sorry to say, I got this problem after the update. Tried to unlock the door, it worked, then to lock the door, I got this message:
> 
> Filed to call service lock/lock. Server returned: ‘Invalid request body - invalid device ID’

Same for me. It works once (e.g. start charge) the second request failes (e.g. stop charge)
Error is also "hyundai_kia_connect_api.exceptions.DeviceIDError: Invalid request body - Invalid deviceId."

**Comment by smalarz** [comment]
same again - ługi kia_uvo/update. Invalid request body - Invalid deviceId.

EU/KIA/NIRO

**Comment by Strauman** [comment]
This exact thing is happening for me when I try to run any service more than once. If I reload the integration it works again 🤔

**Comment by MHultman** [comment]
Error still occurs. 
I get this error when trying to call any service. Works first time after reloading integration, but not for next call and forward. 

```
2023-09-14 23:58:43.524 ERROR (MainThread) [homeassistant.components.websocket_api.http.connection] [546903271872] Invalid request body - Invalid deviceId. 
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/components/websocket_api/commands.py", line 227, in handle_call_service
    await hass.services.async_call(
  File "/usr/src/homeassistant/homeassistant/core.py", line 1969, in async_call
    response_data = await coro
                    ^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/core.py", line 2006, in _execute_service
    return await target(service_call)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/kia_uvo/services.py", line 70, in async_handle_start_climate
    await coordinator.async_start_climate(vehicle_id, climate_request_options)
  File "/config/custom_components/kia_uvo/coordinator.py", line 198, in async_start_climate
    await self.hass.async_add_executor_job(
  File "/usr/local/lib/python3.11/concurrent/futures/thread.py", line 58, in run
    result = self.fn(*self.args, **self.kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/hyundai_kia_connect_api/VehicleManager.py", line 133, in start_climate
    return self.api.start_climate(self.token, self.get_vehicle(vehicle_id), options)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/hyundai_kia_connect_api/KiaUvoApiEU.py", line 835, in start_climate
    _check_response_for_errors(response)
  File "/usr/local/lib/python3.11/site-packages/hyundai_kia_connect_api/KiaUvoApiEU.py", line 113, in _check_response_for_errors
    raise error_code_mapping[response["resCode"]](response["resMsg"])
```

**Comment by plh4** [comment]
Same problem like @MHultman - new version of integration installed. After reload it works to unlock doors. 
Second try - error message "Fehler beim Aufrufen des Diensts lock/unlock. Invalid request body - Invalid deviceId.". 
Same when trying to change the loading limit.

Kia EV6 EU (Germany)
v2.14.1

**Comment by ReidarKri** [comment]
same here, works once then stops with "Invalid deviceId" Kia ENiro Sweden.

**Comment by hmars-t** [comment]
I have exactly the same problem as @MHultman. It works exactly once after setting up the integration. Afterwards I only get the error message "Invalid device ID".

**Comment by crackers81** [comment]
> I have exactly the same problem as @MHultman. It works exactly once after setting up the integration. Afterwards I only get the error message "Invalid device ID".

Same here.
